src/editor/components/scenegraph/SceneGraph.js

```diff
@@ -62,307 +51,63 @@ export default class SceneGraph extends React.Component {
-        this.expandToRoot(entity);
-        if (this.props.onChange) {
-          this.props.onChange(entity);
-        }
-        posthog.capture('entity_selected', {
-          entity: entity.getAttribute('mixin')
-        });
-        Events.emit('entityselect', entity, true);
-        found = true;
-      }
-    }
-
-    if (!found) {
-      this.setState({ selectedEntity: null, selectedIndex: -1 });
-    }
-  };
-
   rebuildEntityOptions = () => {
-    const entities = [{ depth: 0, entity: this.props.scene }];
-
-    function treeIterate(element, depth) {
-      if (!element) {
-        return;
-      }
-      depth += 1;
-
-      for (let i = 0; i < element.children.length; i++) {
-        let entity = element.children[i];
-
-        if (
-          entity.dataset.isInspector ||
-          !entity.isEntity ||
-          entity.isInspector ||
-          'aframeInspector' in entity.dataset
-        ) {
-          continue;
-        }
-
-        entities.push({ entity: entity, depth: depth });
-
-        treeIterate(entity, depth);
-      }
-    }
-    treeIterate(this.props.scene, 0);
-
     this.setState({
-      entities: entities,
-      filteredEntities: this.getFilteredEntities(this.state.filter, entities)
+      scene: this.props.scene
     });
   };
 
-  selectIndex = (index) => {
-    if (index >= 0 && index < this.state.entities.length) {
-      this.selectEntity(this.state.entities[index].entity);
-    }
-  };
-
-  onFilterKeyUp = (event) => {
-    if (event.keyCode === 27) {
-      this.clearFilter();
-    }
-  };
-
-  onKeyDown = (event) => {
-    switch (event.keyCode) {
-      case 37: // left
-      case 38: // up
-      case 39: // right
-      case 40: // down
-        event.preventDefault();
-        event.stopPropagation();
-        break;
-    }
-  };
-
-  onKeyUp = (event) => {
-    if (this.props.selectedEntity === null) {
-      return;
-    }
-
-    switch (event.keyCode) {
-      case 37: // left
-        if (this.isExpanded(this.props.selectedEntity)) {
-          this.toggleExpandedCollapsed(this.props.selectedEntity);
-        }
-        break;
-      case 38: // up
-        this.selectIndex(
-          this.previousExpandedIndexTo(this.state.selectedIndex)
-        );
-        break;
-      case 39: // right
-        if (!this.isExpanded(this.props.selectedEntity)) {
-          this.toggleExpandedCollapsed(this.props.selectedEntity);
-        }
-        break;
-      case 40: // down
-        this.selectIndex(this.nextExpandedIndexTo(this.state.selectedIndex));
-        break;
-    }
-  };
-
-  getFilteredEntities(filter, entities) {
-    entities = entities || this.state.entities;
-    if (!filter) {
-      return entities;
-    }
-    return entities.filter((entityOption) => {
-      return filterEntity(entityOption.entity, filter || this.state.filter);
-    });
-  }
-
-  isVisibleInSceneGraph = (x) => {
-    let curr = x.parentNode;
-    if (!curr) {
+  shouldRenderNode = (node) => {
```

Comment:
can add to this to determine what to render in the main scene graph. Notice how we're hiding things in `HIDDEN_CLASSES` like`teleportRay`